.script/tests/KqlvalidationsTests/DetectionsYamlFilesTestData.cs

```diff
@@ -0,0 +1,62 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+using System.Text;
+
+namespace Kqlvalidations.Tests
+{
+    public class DetectionsYamlFilesTestData : TheoryData<string>
+    {
+        public DetectionsYamlFilesTestData()
+        {
+            string detectionPath = GetDetectionPath();
+            var files = Directory.GetFiles(detectionPath, "*.yaml", SearchOption.AllDirectories).ToList();
+            files.ForEach(f => AddData(Path.GetFileName(f)));
+        }
+
+        public static string GetDetectionPath()
+        {
+            var rootDir = Directory.CreateDirectory(GetAssemblyDirectory());
+            var testFolderDepth = 6;
+            for (int i = 0; i < testFolderDepth; i++)
+            {
+                rootDir = rootDir.Parent;
+            }
+            var detectionPath = Path.Combine(rootDir.FullName, "Detections");
+            return detectionPath;
+        }
+
+        public static string GetCustomTablesPath()
+        {
+            var rootDir = Directory.CreateDirectory(GetAssemblyDirectory());
+            var testFolderDepth = 3;
```

Comment:
What happened in case someone change the folder organizetion?
are we going to see a message about incorrect path?

Reply:
yes but why should it happen